src/Abp.EntityFrameworkCore.EFPlus/EntityFrameworkCore/EFPlus/AbpEntityFrameworkCoreEFPlusExtensions.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Domain.Repositories;
+using Abp.Domain.Uow;
+using Abp.Linq.Expressions;
+using Abp.Runtime.Session;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Z.EntityFramework.Plus;
+
+namespace Abp.EntityFrameworkCore.EFPlus
+{
+    /// <summary>
+    /// Defines batch delete and update extension methods for IRepository
+    /// </summary>
+    public static class AbpEntityFrameworkCoreEfPlusExtensions
+    {
+        /// <summary>
+        /// Deletes all matching entities permanently for given predicate
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <typeparam name="TPrimaryKey">Primary key type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchDeleteAsync<TEntity, TPrimaryKey>([NotNull] this IRepository<TEntity, TPrimaryKey> repository, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<TPrimaryKey>
+        {
+            Check.NotNull(repository, nameof(repository));
+
+            var query = repository.GetAll().IgnoreQueryFilters();
+
+            var abpFilterExpression = GetFilterExpressionOrNull<TEntity, TPrimaryKey>(repository.GetIocResolver());
+            var filterExpression = ExpressionCombiner.Combine(predicate, abpFilterExpression);
+
+            if (filterExpression != null)
```

Comment:
filterExpression should never be null (assuming the predicate is not null). So throw exception instead of deleting all table!